GVFS/GVFS.Common/NuGetUpgrader/NuGetUpgrader.cs

```diff
@@ -144,6 +151,30 @@ public class NuGetUpgrader : ProductUpgrader
             return true;
         }
 
+        public static bool TryCreateAzDevOrgUrlFromPackageFeedUrl(string packageFeedUrl, out string azureDevOpsUrl, out string error)
+        {
+            // We expect a URL of the form https://pkgs.dev.azure.com/{org}
+            // and want to convert it to a URL of the form https://dev.azure.com/{org}
+            Regex packageUrlRegex = new Regex(
+                @"^https://pkgs.dev.azure.com/(?<org>.+?)/",
+                RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
+            Match urlMatch = packageUrlRegex.Match(packageFeedUrl);
+
+            if (!urlMatch.Success)
+            {
+                azureDevOpsUrl = null;
+                error = $"Input URL {packageFeedUrl} did not match expected format for an Azure Devops Package Feed URL";
+                return false;
+            }
+
+            string org = urlMatch.Groups["org"].Value;
+
+            azureDevOpsUrl = urlMatch.Result($"https://dev.azure.com/{org}");
```

Comment:
I tried this out on my actual machine, and it seems that the credential manager does not like this format. I can use `https://{org}.visualstudio.com`, but not `https://dev.azure.com/{org}`. I will take a closer look at what might be going on here.

/cc @mjcheetha, @jrbriggs m in case I am missing something obvious with URLs and GCM

Reply:
What do you mean by "does not like this format"? Are you seeing an error? Is this Windows GCM?

For Windows GCM we don't expect/work with the 'useHttpPath' option, but instead do that process walking logic to find the git-remote-helper/http process, which VFS for Git won't have as part of the process tree.

Reply:
You can use `https://{org}@dev.azure.com/{org}` for maximum compat.

Reply:
ah - yeah, that is it. I forgot that was how the Windows GCM worked.